#!/usr/bin/env python
PACKAGE = "tello_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add( name , type , reconfiguration_lvl , description , default_value ,  min , max )

# X variables
gen.add("x_d", double_t, 0, "X desired value", 0.0, -1.0, 1.0)
gen.add("kp_x", double_t, 0, "X proportional gain", 0.5, 0.0, 5.0)
gen.add("kd_x", double_t, 0, "X derivative gain", 0.05, 0.0, 1.0)

# Y variables
gen.add("y_d", double_t, 0, "Y desired value", 0.0, -1.0, 1.0)
gen.add("kp_y", double_t, 0, "Y proportional gain", 0.5, 0.0, 5.0)
gen.add("kd_y", double_t, 0, "Y derivative gain", 0.05, 0.0, 1.0)

# Z variables
gen.add("z_d", double_t, 0, "Z desired value", 1.0, 0.5, 3.0)
gen.add("kp_z", double_t, 0, "Z proportional gain", 0.5, 0.0, 5.0)
gen.add("kd_z", double_t, 0, "Z derivative gain", 0.05, 0.0, 1.0)

# Yaw variables
gen.add("yaw_d", int_t, 0, "Yaw desired value", 0, -180, 180)
gen.add("kp_yaw", double_t, 0, "Yaw proportional gain", 0.5, 0.0, 3.0)
gen.add("kd_yaw", double_t, 0, "Yaw derivative gain", 0.05, 0.0, 1.0)

exit(gen.generate(PACKAGE, "tello_control", "dynamic_gains"))
